sdk/src/main/java/com/qonversion/android/sdk/internal/user/controller/UserControllerImpl.kt

```diff
@@ -0,0 +1,61 @@
+package com.qonversion.android.sdk.internal.user.controller
+
+import androidx.annotation.VisibleForTesting
+import com.qonversion.android.sdk.dto.User
+import com.qonversion.android.sdk.internal.cache.CacheState
+import com.qonversion.android.sdk.internal.cache.Cacher
+import com.qonversion.android.sdk.internal.common.BaseClass
+import com.qonversion.android.sdk.internal.common.StorageConstants
+import com.qonversion.android.sdk.internal.exception.ErrorCode
+import com.qonversion.android.sdk.internal.exception.QonversionException
+import com.qonversion.android.sdk.internal.logger.Logger
+import com.qonversion.android.sdk.internal.user.service.UserService
+
+internal class UserControllerImpl(
+    private val userService: UserService,
+    private val userCacher: Cacher<User>,
+    logger: Logger
+) : UserController, BaseClass(logger) {
+
+    override suspend fun getUser(): User {
+        logger.verbose("getUser() -> started")
+
+        val cachedUserDefault = userCacher.getActual(StorageConstants.UserInfo.key)
+        if (cachedUserDefault != null) {
+            return cachedUserDefault
+        }
+        try {
+            val userId = userService.obtainUserId()
+            val user = userService.getUser(userId)
+            logger.info("User info was successfully received from API")
+
+            storeUser(user)
+            return user
+        } catch (exception: QonversionException) {
+            val cachedUserError =
```

Comment:
Let's rename this value to `cachedUser` because cachedUserError looks like an error object name